### PR TITLE
Docs: explain how Composer gets the stability from a version

### PR DESCRIPTION
**Pull request by glaubinix**
I couldn't find a place in the docs that explains what makes a stable version stable or not stable.

### CODE REVIEW

File: doc/articles/versions.md [hunk 1]
```diff
@@ -80,6 +80,19 @@ In the above example, if you wanted to check out the `my-feature` branch, you wo
...
+Composer recognizes the following stabilities (in order of stability): dev,
+alpha, beta, RC, and stable where RC stands for release candidate. The stability
+of a version is defined by its suffix e.g version `v1.1-BETA` has a stability of
+`beta` and `v1.1-RC1` has a stability of `rc`. If such a suffix is missing
+e.g. version `v1.1` then Composer considers that version `stable`. In addition
+to that Composer automatically adds a `-dev` suffix to all numeric branches and
+prefixes all other branches imported from a VCS repository with `dev-`. In both
+cases the stability `dev` gets assigned.
+
+Keeping this in mind will help us in the next section.
```
Comment (stof) [inline doc/articles/versions.md:94]: this sentence does not fit the tone of documentation, which does not use `us`

File: doc/articles/versions.md [hunk 2]
```diff
@@ -80,6 +80,19 @@ In the above example, if you wanted to check out the `my-feature` branch, you wo
 
 When branch names look like versions, we have to clarify for composer that we're trying to check out a branch and not a tag. In the above example, we have two version branches: `v1` and `v2`. To get Composer to check out one of these branches, you must specify a version constraint that looks like this: `v1.x-dev`. The `.x` is an arbitrary string that Composer requires to tell it that we're talking about the `v1` branch and not a `v1` tag (alternatively, you can name the branch `v1.x` instead of `v1`). In the case of a branch with a version-like name (`v1`, in this case), you append `-dev` as a suffix, rather than using `dev-` as a prefix.
 
+### Stabilities
+
+Composer recognizes the following stabilities (in order of stability): dev,
+alpha, beta, RC, and stable where RC stands for release candidate. The stability
```
Comment (stof) [inline doc/articles/versions.md:86]: these doc files are not using hard wrapping currently

Comment (glaubinix) [reply doc/articles/versions.md:86]: They do it's just that the section before and after this do not

File: doc/articles/versions.md (hunk 1 above)
Comment (glaubinix) [reply doc/articles/versions.md]: Updated

File: doc/articles/versions.md [hunk 3]
```diff
@@ -80,6 +80,19 @@ In the above example, if you wanted to check out the `my-feature` branch, you wo
...
+of a version is defined by its suffix e.g version `v1.1-BETA` has a stability of
+`beta` and `v1.1-RC1` has a stability of `rc`. If such a suffix is missing
```
Comment (Seldaek) [inline doc/articles/versions.md:88]:
```suggestion
`beta` and `v1.1-RC1` has a stability of `RC`. If such a suffix is missing
```

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**